components/greaselion/browser/greaselion_service_impl.cc

```diff
@@ -282,29 +280,33 @@ void GreaselionServiceImpl::CreateAndInstallExtensions() {
         rule->has_unknown_preconditions() == false) {
       // Convert script file to component extension. This must run on extension
       // file task runner, which was passed in in the constructor.
+      GreaselionRule rule_copy(*rule);
       base::PostTaskAndReplyWithResult(
           task_runner_.get(), FROM_HERE,
           base::BindOnce(&ConvertGreaselionRuleToExtensionOnTaskRunner,
-                         rule.get(), install_directory_, &extension_dirs_),
+                         rule_copy, install_directory_),
           base::BindOnce(&GreaselionServiceImpl::PostConvert,
                          weak_factory_.GetWeakPtr()));
     }
   }
 }
 
 void GreaselionServiceImpl::PostConvert(
-    scoped_refptr<extensions::Extension> extension) {
-  if (!extension.get()) {
+    std::unique_ptr<GreaselionConvertedExtension> converted_extension) {
+  if (!converted_extension) {
     all_rules_installed_successfully_ = false;
     pending_installs_ -= 1;
     MaybeNotifyObservers();
     LOG(ERROR) << "Could not load Greaselion script";
   } else {
-    greaselion_extensions_.push_back(extension->id());
+    auto extension_dir = std::move(converted_extension->extension_dir());
+    extension_dirs_.push_back(std::move(*extension_dir));
```

Comment:
Nit: could maybe just do:

```cpp
extension_dirs_.push_back(std::move(converted_extension->extension_dir()));
```

Reply:
This line ended up gettting factored out.